lib/klaviyo/client.rb

```diff
@@ -1,24 +1,185 @@
 require 'open-uri'
 require 'base64'
 require 'json'
+require 'Faraday'
 
 module Klaviyo
   class KlaviyoError < StandardError; end
-  
+
+  NO_PRIVATE_API_KEY_ERROR = 'Please provide Private API key for this request'
+
   class Client
-    def initialize(api_key, url = 'http://a.klaviyo.com/')
+    def initialize(api_key = nil, private_api_key = nil, domain = 'https://a.klaviyo.com/api')
+      if !api_key
+        KlaviyoError.new('Please provide your Public API key')
+      end
+
       @api_key = api_key
-      @url = url
+      @private_api_key = private_api_key
+      @domain = domain
+
+      if @private_api_key
+        @private_api_key_param = "api_key=#{@private_api_key}"
+      end
+
+      @v1 = 'v1'
+      @v2 = 'v2'
+      @metric = 'metric'
+      @metrics = 'metrics'
+      @timeline = 'timeline'
+      @list = 'list'
+      @lists = 'lists'
+      @person = 'person'
+      @subscribe = 'subscribe'
+      @members = 'members'
+      @exclusions = 'exclusions'
+      @all = 'all'
+      @group = 'group'
+
+    end
+
+# METRICS API
+
+    def get_metrics(kwargs = {})
+      path = @metrics
+
```

Comment:
I think you have way to many empty lines her, please take a look at https://github.com/rubocop-hq/ruby-style-guide
```suggestion
```